src/keyValueStore.js

```diff
@@ -17,9 +17,22 @@ class KeyValueStore {
    * @param     {String}    key                     the key
    * @return    {String}                            the value associated with the key, undefined if there's no such key
    */
-  async get (key) {
+  async get (key, opts = {}) {
     const x = await this._get(key)
-    return x ? x.value : x
+
+    if (!x) {
+      return x
+    }
+
+    if (opts.metadata) {
+      const metadata = this._extractMetadata(x)
+      return {
+        ...metadata,
+        value: x.value
+      }
+    }
+
+    return x.value
```

Comment:
This is a bit funky but I assumed that you'd want keep compatibility with the old api returning a strings when the `metadata` option is not passed, is that a correct assumption?

Reply:
Hm, I can't remember exactly why we had `x ? x.value : x` rather than just returning null if x is not defined. I think this is clear enough though!

Reply:
> Cool, Thanks for picking this up @kimpers!
> 
> Looks good, but I think this could have some side effects that needs to be addressed. Specifically in `publicStore` and `privateStore` classes, as well as the public/private reducers in the `space` class.

I think everything should be addressed now